test/operance_data_controller_test.dart

```diff
@@ -0,0 +1,444 @@
+        );
+      });
+    });
+
+    group('When getting visible columns', () {
+      test('Then it should return the visible columns', () async {
+        await initializeController(
+          columnOrder: columnOrder,
+          initialPage: initialPageWithRows,
+        );
+
+        controller.hideColumn(1);
+        expect(controller.visibleColumns, <int>{0, 2});
+      });
+    });
+
+    group('When hiding and showing columns', () {
+      test('Then it should hide and show columns correctly', () async {
+        await initializeController(
+          columnOrder: columnOrder,
+          initialPage: initialPageWithRows,
+        );
+
+        controller.hideColumn(1);
+        expect(controller.hiddenColumns, <int>{1});
+
+        controller.showColumn(1);
+        expect(controller.hiddenColumns, isEmpty);
+      });
+
+      test('Then it should toggle column visibility', () async {
+        await initializeController(
+          columnOrder: columnOrder,
+          initialPage: initialPageWithRows,
+        );
+
+        controller.toggleColumnVisibility(1);
+        expect(controller.hiddenColumns, <int>{1});
+
+        controller.toggleColumnVisibility(1);
+        expect(controller.hiddenColumns, isEmpty);
+      });
+
+      test('Then it should show all columns', () async {
+        await initializeController(
+          columnOrder: columnOrder,
+          initialPage: initialPageWithRows,
+        );
+
+        controller
+          ..hideColumn(1)
+          ..showAllColumns();
+        expect(controller.hiddenColumns, isEmpty);
+      });
+    });
+
+    group('When managing searched rows', () {
+      test('Then it should clear searched rows', () async {
+        await initializeController(
+          initialPage: initialPageWithRows,
+        );
+
+        controller
+          ..addSearchedRows(<String>['Row1'])
+          ..clearSearchedRows();
+        expect(controller.searchedRows, isEmpty);
+      });
+
+      test('Then it should add searched rows', () async {
+        await initializeController(
+          initialPage: initialPageWithRows,
+        );
+
+        controller.addSearchedRows(<String>['Row1']);
+        expect(controller.searchedRows, <String>{'Row1'});
+      });
+    });
+
+    group('When setting rows per page', () {
+      test('Then it should set the number of rows per page', () async {
+        await initializeController(
+          initialPage: initialPageWithRows,
+        );
+
+        controller.setRowsPerPage(50);
+        expect(controller.rowsPerPage, 50);
+      });
+    });
+
+    group('When reordering columns', () {
+      test('Then it should reorder columns', () async {
+        await initializeController(
+          columnOrder: columnOrder,
+          initialPage: initialPageWithRows,
+        );
+
+        controller.reOrderColumn(0, 2);
+        expect(controller.columnOrder, [1, 2, 0]);
+      });
+    });
+
+    group('When fetching data', () {
```

Comment:
The infinite scroll functionality is initialised in some tests but there aren't many specific tests for infinite scroll behaviour.

Maybe more tests for:

Scroll threshold behavior
Loading states during infinite scroll
Error handling during infinite scroll fetches